rosidl_runtime_py/rosidl_runtime_py/convert.py

```diff
@@ -85,41 +92,67 @@ def to_string(val):
                         val += '...'
             r = str(val)
         else:
-            r = message_to_csv(val, truncate_length)
+            r = message_to_csv(val, truncate_length, no_arr, no_str)
         return r
     result = ''
     # We rely on __slots__ retaining the order of the fields in the .msg file.
     for field_name in msg.__slots__:
         value = getattr(msg, field_name)
+        field_type = msg._fields_and_field_types[field_name[1:]]
+
         if result:
             result += ','
-        result += to_string(value)
+
+        if no_arr is True and any(
+                isinstance(value, t) for t in [list, tuple, array.array, numpy.ndarray]):
```

Comment:
@vinnamkim I _think_ that you may achieve the same behavior by just doing:

```python
isinstance(value, (list, tuple, array.array, numpy.ndarray))
```

Reply:
Fixed.